docs/bootloader/runner_flow.md

```diff
@@ -6,57 +6,26 @@ This section describes how the bootloader interacts with the execution environme
 
 The bootloader implements (and uses) two entrypoints for code execution.
 
-The first one is `run_till_completion` from the [`runner`](../../basic_bootloader/src/bootloader/runner.rs) module of the bootloader. This function implements the main execution loop given an initial request (either external call or deployment), which is explained in the next section.
+The first one is `run_till_completion` from the [`runner`](../../basic_bootloader/src/bootloader/runner.rs) module of the bootloader. This function implements the main execution loop given an initial call request, which is explained in the next section.
 
  The second one, [`run_single_interaction`](../../basic_bootloader/src/bootloader/run_single_interaction.rs), is just a simple wrapper over the previous to simplify external calls from the bootloader. It just adds the logic for starting and finishing the topmost execution frame and prepares the inputs for `run_till_completion`.
 
 ## Runner structure
 
 The runner's responsibility is to coordinate calls into the execution environments. For this, the runner keeps a callstack of execution environment states and will be responsible of starting and finishing system frames. Frames are used to take snapshots for storage and memory to which the system can revert to in case of a failure.
 
-The runner is implemented as an infinite loop that dispatches the spawn requests returned by the execution environment. As a reminder, these are:
-
-- External call request and
-- Deployment request.
+The runner is implemented as an infinite loop that dispatches the spawn requests returned by the execution environment. As a reminder, these are used both for external calls and execution of constructor code for deployments.
 
 The runner breaks out of the infinite loop after processing the completion of the initial request (when the callstack becomes empty).
 
 ### Call request
 
 For the external call request, the bootloader needs to:
```

Comment:
Note: I think in future we should split concepts of "bootloader" and "runner"

Reply:
Hmm interesting, so the runner would be a 4th component? Why?